src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -295,11 +296,11 @@ export default function Swap({
       } catch (error) {
         // try to approve if gatherPermitSignature failed for any reason other than the user rejecting it
         if (error?.code !== 4001) {
-          await approveCallback()
+          await applyCustomGasPrice(approveCallback)
```

Comment:
should we add the cutom gas price already inside the `useApproveCallback` hook? 

my concern would be, if we use `useApproveCallback` in several places, we would need to replicate this logic over and over, and it would be easy to miss...